generator/api/src/main/java/io/fabric8/maven/generator/api/support/AbstractPortsExtractor.java

```
@@ -0,0 +1,178 @@
+        String configPath = System.getProperty(getConfigPathPropertyName(), getConfigPathFromProject(project));
+        if (configPath == null) {
+            return null;
+        }
+        return Paths.get(configPath).toFile();
+    }
+
+    @Override
+    public Map<String, Integer> extract(MavenProject project) {
+        Map<String, Integer> answer = new HashMap<>();
+        File configFile = getConfigLocation(project);
+        if (!configFile.exists()) {
+            log.warn("Could not find config: [%s]. Ignoring.", configFile.getAbsolutePath());
+            return null;
+        }
+
+        try {
+            Map<String, String> configMap = readConfig(configFile);
+            for (Map.Entry<String, String> entry : configMap.entrySet()) {
+                String key = entry.getKey();
+                if (isValidPortPropertyKey(key)) {
+                    addPortIfValid(answer, key, entry.getValue());
+                }
+            }
+        } catch (IOException e) {
+            log.warn("Error reading config: [%s], due to: [%s]. Ignoring.", configFile.getAbsolutePath(), e.getMessage());
+        }
+        return answer;
+    }
+
+
+
+    /**
+     * Reads the configuration from the file.
+     * @param f
+     * @return
+     * @throws IOException
+     */
+    static Map<String, String> readConfig(File f) throws IOException {
+        Map<String, String> map;
+        if (f.getName().endsWith(JSON_EXTENSION)) {
+            map = flatten(JSON_MAPPER.readValue(f, Map.class));
+        } else if (f.getName().endsWith(YAML_EXTENSION) || f.getName().endsWith(YML_EXTENSION)) {
+            map = flatten(YAML_MAPPER.readValue(f, Map.class));
+        } else if (f.getName().endsWith(PROPERTIES_EXTENSION)) {
+            Properties properties = new Properties();
+            properties.load(new FileInputStream(f));
+            map = propertiesToMap(properties);
+        } else {
+            map = Collections.emptyMap();
```

Maybe better to throw an exception here, that the config format (like 'toml' as Vert.x is starting to look in) is not supported, instead of silently failing to extract a port from a given file name.